### DISCUSSION TITLE
Is Yaru missing document-new-symbolic?

### DISCUSSION THREAD

**Discussion by CraigFoote** [ubuntu/yaru#3999] [General]
Fully up-to-date Ubuntu 23.04. I have an app in Builder that, when run, displays a warning symbol on a GtkButton declared in .ui with ‘icon-name’ set to ‘**document-new-symbolic**’. When I switch to Adwaita icon theme in Tweaks, it displays properly. Switching back to Yaru brings back the warning symbol. I do see the icon in `/usr/share/icons/Adwaita/scalable/actions` and in the `Yaru` folder but they're different sizes. Am I correct in assuming the problem is in a .theme file somewhere? Any ideas? Beautiful theme BTW.

```
            .-/+oossssoo+/-.               craig@notebook 
        `:+ssssssssssssssssss+:`           -------------- 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 23.04 x86_64 
    .ossssssssssssssssssdMMMNysssso.       Host: XPS 13 9310 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 6.2.0-33-generic 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 55 mins 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 2758 (dpkg), 44 (flatpak), 25 (snap) 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: bash 5.2.15 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Resolution: 3456x2160 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   DE: GNOME 44.3 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: Mutter 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   WM Theme: Adwaita 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Theme: Yaru-sage-dark [GTK2/3] 
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Icons: Yaru-sage-dark [GTK2/3] 
  +sssssssssdmydMMMMMMMMddddyssssssss+     Terminal: gnome-terminal 
   /ssssssssssshdmNNNNmyNMMMMhssssss/      CPU: 11th Gen Intel i7-1195G7 (8) @ 5.000GHz 
    .ossssssssssssssssssdMMMNysssso.       GPU: Intel TigerLake-LP GT2 [Iris Xe Graphics] 
      -+sssssssssssssssssyyyssss+-         Memory: 3785MiB / 15692MiB 
        `:+ssssssssssssssssss+:`
            .-/+oossssoo+/-.
```

**Comment by jbicha** [comment]
What is the exact version of GNOME Builder you are using?

Is the code for the app you're working on public?

**Comment by CraigFoote** [comment]
Builder v45.1 (nightly). No I'm sorry, the code is not public.

**Reply by jbicha** [reply]
Ok, so you're using a Flatpak version of Builder, right?

Do you have the Yaru Flatpak theme? What version?

Is your app built as a Flatpak?

**Comment by CraigFoote** [comment]
I made the project using the Builder wizard for a new "Gnome Application" and I do believe it builds into a flatpak but I'm not certain what happens when launching it from within Builder like I've been doing. Anyway how do I tell if I have Yaru Flatpak installed and at which version. Sorry, I'm a bit of a noob :)

**Comment by CraigFoote** [comment]
I just got this from flatpak list:
`Yaru-sage-dark Gtk Theme               org.gtk.Gtk3theme.Yaru-sage-dark                      3.22        flathub       system`

**Comment by CraigFoote** [comment]
@jbicha I tested with Ubuntu-23.10-beta, again with Builder Nightly flatpak, and the icon again failed to render with Yaru theme and again worked properly with Adwaita theme. `flatpak list` revealed I do not have the Yaru flatpak installed, not sure where to find it or if it would make a difference.

**Comment by CraigFoote** [comment]
@jbicha Should https://github.com/ubuntu/yaru/tree/master/icons/Yaru/scalable/actions have the document-new-symbolic.svg?

**Reply by Jupi007** [reply]
Hi @CraigFoote,
we already have the `document-new-symbolic` icon: https://github.com/ubuntu/yaru/blob/master/icons/Yaru/scalable/actions/document-new-symbolic.svg

**Reply by CraigFoote** [reply]
@jbicha Hmm, not sure how I missed that, sorry. Still not getting the icon on 23.04 or 23.10 tho :(

**Reply by Jupi007** [reply]
I don't think it comes from an outdated flatpak yaru theme, because we have this icon since the beginning of this repository 🤔
Honestly, I think it is not related to yaru itself.

Did you tried another third party icon theme?

**Comment by CraigFoote** [comment]
No I've only tested with Yaru and Adwaita. New install coming in a few days but I don't expect it to behave any different than my 23.10-beta in virtualbox.

**Reply by Jupi007** [reply]
Can you try another theme, like Numix?

**Reply by CraigFoote** [reply]
Will do...

**Reply by CraigFoote** [reply]
Works with Numix icon theme.

**Reply by Jupi007** [reply]
Hmm, okay thanks for testing.

I found this icon in the `gtk4-widget-factory`:

![image](https://github.com/ubuntu/yaru/assets/36476595/e9719bd6-f5dd-4014-8295-ac98943aa574)

So it should be related to the flatpak packaging of yaru, but not to yaru itself.

@Muqtxdir Have you any thought about this?

**Reply by CraigFoote** [reply]
I don't have the yaru flatpak installed according to `flatpak list`.

**Reply by Jupi007** [reply]
Oh, well did you tried to install it?

**Reply by CraigFoote** [reply]
No, how exactly do I do that?

**Reply by CraigFoote** [reply]
I do have the yaru flatpak installed in 23.04, just not 23.10.

**Reply by CraigFoote** [reply]
I didn't explicitly install it on 23.04 BTW.

**Reply by CraigFoote** [reply]
Code is at https://github.com/CraigFoote/ca.footeware.c.texty if it helps. Not much to it, mostly commented out from a tutorial I was reading.

**Reply by Jupi007** [reply]
I'm sorry, I've never used gtk. So I cannot help you with that.

As I am not a flatpak user, your problem seems to be beyond my knowledge.

**Reply by CraigFoote** [reply]
Any idea where I could forward this discussion? Flatpak devs or something.

**Reply by Jupi007** [reply]
Maybe @jbicha or @Muqtxdir can help you.

**Reply by CraigFoote** [reply]
OK, thx for looking tho.

**Reply by Muqtxdir** [reply]
hello, @CraigFoote 

The apps built and run using `gnome-builder` reply on runtimes from flatpak repo. So the icons it picks up from are from `hicolor` dir either from the `Sdk` and `Platform` runtimes not from the choosen host icon theme.

I'll reach home in an hour, I'll give look after that to this discussion.

**Reply by CraigFoote** [reply]
Ah the light just went on, thx. OK, I'll be around if I can help further.
